### PR TITLE
Pom and Javadoc cleanup for sonatype deploy.

### PR DESCRIPTION
**Pull request by tedcasey**
Changes required in order to comply with sonatype deploy requirements.

Added pom entries to generate source and javadoc jars
Added pom information required by sonatype - license, developers, scm info.
Cleaned up Javadocs per the errors generated by the maven javadocs plugin

## Motivation and Context

Want to deploy to sonatype.

## How Has This Been Tested?

Compiled all libraries and verified javadoc and source jars are created.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: core/src/main/java/com/adobe/ride/core/controllers/RestApiController.java [hunk 1]
```diff
@@ -102,10 +102,10 @@ public static RequestSpecBuilder getRequestBuilder(boolean useDefaultHeaders, Fi
    * Utility method which returns a prepared RequestSpecBuilder to the API calls if the caller
    * hasn't passed one
    *
-   * @param requestSpec
-   * @param object
-   * @param method
-   * @return
+   * @param requestSpec Rest-assured definitions for the call.
+   * @param object name of the target REST object, usually a segment in the path
```
Comment (enortonAtAdobe) [inline core/src/main/java/com/adobe/ride/core/controllers/RestApiController.java]: is this the correct description? elsewhere, the description has been "ModelObject used in conjunction with the call."

File: core/src/main/java/com/adobe/ride/core/controllers/RestApiController.java [hunk 2]
```diff
@@ -61,8 +61,8 @@
    * Use to ensure logging is always enabled for RA requests.
    *
    * @param useDefaultHeaders If true, enables default content-type, accept, and client headers.
-   * @param filter The custom Rest-Assured filter to apply to the request.
-   * @return A static reference to a new {@link RequestSpecBuilder} instance.
+   * @param filters The custom Rest-Assured filter to apply to the request.
```
Comment (enortonAtAdobe) [inline core/src/main/java/com/adobe/ride/core/controllers/RestApiController.java]: update to 'filters Rest-Assured filters to be used in the call' (plural from singular description)

File: core/src/main/java/com/adobe/ride/core/filters/TestNGReporterLogFilter.java [hunk 3]
```diff
@@ -97,10 +96,10 @@ private String getHeaderString(Headers headers) {
   }
 
   /**
-   * Formats a variety of request params for reporting.
+   * construction method for getting a serialized list of the the headers used in the call.
```
Comment (enortonAtAdobe) [inline core/src/main/java/com/adobe/ride/core/filters/TestNGReporterLogFilter.java]: params, not headers
also, perhaps should be more specific about getting the values of the params

File: libraries/ride-fuzzer-lib/src/main/java/com/adobe/ride/libraries/fuzzer/HeaderFuzzer.java [hunk 4]
```diff
@@ -155,8 +160,8 @@ public void fuzzHeaderWithNoSQLInjectionStrings(String scope, String header, Obj
   /**
    * Method to validate a 4xx error responses for expected failures and 2xx expected successes.
    * 
-   * @param response
-   * 
+   * @param response Rest-Assured Response
+   * @param expectSuccess boolean that indicate whether the call should pass or fail
```
Comment (enortonAtAdobe) [inline libraries/ride-fuzzer-lib/src/main/java/com/adobe/ride/libraries/fuzzer/HeaderFuzzer.java]: should be indicates (with an 's')

File: utilities/ride-model-util/src/main/java/com/adobe/ride/utilities/model/ModelObject.java [hunk 5]
```diff
@@ -1379,7 +1384,7 @@ protected String getRelativeResourceLocation(String controlPath, String relative
   }
 
   /**
-   * method to sync the JSONObject type objectMetadata
+   * method to sync the JSONObject type objectMetadata.
```
Comment (enortonAtAdobe) [inline utilities/ride-model-util/src/main/java/com/adobe/ride/utilities/model/ModelObject.java]: Capitalize method

**Review by enortonAtAdobe — APPROVED** [review]

File: core/src/main/java/com/adobe/ride/core/controllers/RestApiController.java (hunk 1 above)
Comment (tedcasey) [reply core/src/main/java/com/adobe/ride/core/controllers/RestApiController.java]: Corrected.

File: core/src/main/java/com/adobe/ride/core/controllers/RestApiController.java (hunk 2 above)
Comment (tedcasey) [reply core/src/main/java/com/adobe/ride/core/controllers/RestApiController.java]: Corrected.

File: core/src/main/java/com/adobe/ride/core/filters/TestNGReporterLogFilter.java (hunk 3 above)
Comment (tedcasey) [reply core/src/main/java/com/adobe/ride/core/filters/TestNGReporterLogFilter.java]: Correctedl

File: libraries/ride-fuzzer-lib/src/main/java/com/adobe/ride/libraries/fuzzer/HeaderFuzzer.java (hunk 4 above)
Comment (tedcasey) [reply libraries/ride-fuzzer-lib/src/main/java/com/adobe/ride/libraries/fuzzer/HeaderFuzzer.java]: Corrected.

File: utilities/ride-model-util/src/main/java/com/adobe/ride/utilities/model/ModelObject.java (hunk 5 above)
Comment (tedcasey) [reply utilities/ride-model-util/src/main/java/com/adobe/ride/utilities/model/ModelObject.java]: Corrected.

**Pull request merged**